backend/msgpack/writer_test.go

```diff
@@ -282,3 +382,80 @@ type mockInstanceQueue struct {
 
 func (mq *mockInstanceQueue) Enqueue(buf msgpack.Buffer) error { return mq.enqueueFn(buf) }
 func (mq *mockInstanceQueue) Close() error                     { return nil }
+
+type mockSerialInstanceWriter struct {
+	sync.Mutex
+
+	flushSize   int
+	encoder     msgpack.UnaggregatedEncoder
+	encoderPool msgpack.BufferedEncoderPool
+	queue       instanceQueue
+}
+
+func (mw *mockSerialInstanceWriter) Flush() error { return nil }
+func (mw *mockSerialInstanceWriter) Close() error { return nil }
+
+func (mw *mockSerialInstanceWriter) Write(
```

Comment:
Instead of copying and pasting, you could also embed a writer and call `encodeWithLock` directly?